File: codes/OrdenamientoTopologico.cpp
#include<bits/stdc++.h>
#define endl '\n'

using namespace std;

int vertices, aristas, a, b;
vector<int> grafo[10005];
int indegree[10005];
int resp[10005];

bool orden_topologico()
{
    int cont = 0;
    for(int i = 1; i <= vertices; i++)
    {
        if (indegree[i] == 0)
        {
            resp[cont] = i;
            cont++;
            indegree[i]--;
            for (int j = 0; j < grafo[i].size(); j++)
            {
                indegree[grafo[i][j]]--;
            }
            i = 0;
        }
    }

    if (cont == vertices)
        return true;
    else
        return false;
}

int main()
{
    ios_base :: sync_with_stdio(0);
    cin.tie(0);

    cin >> vertices >> aristas;

    for(int i = 0; i < aristas; i++)
    {
        cin >> a >> b;

        grafo[a].push_back(b);
        indegree[b] += 1;
    }

    if(orden_topologico())
    {
        cout << "True" << endl;
        for(int i = 0; i < vertices; i++)
        {
            cout << resp[i] << " ";
        }
        cout << endl;
    }
    else
        cout << "False" << endl;
}

File: codes/OrdenamientoTopologico.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace OrednamientoTopologico
{
    class Program
    {
        static int[] ordenTopologico;

        static bool OrdenTopologico(List<int>[] grafo, int[] indegree)
        {
            ordenTopologico = new int[grafo.Length];
            int cont = 0;

            for (int i = 0; i < indegree.Length; i++)
            {
                if (indegree[i] == 0)
                {
                    ordenTopologico[cont] = i;
                    cont++;
                    indegree[i]--;
                    for (int j = 0; j < grafo[i].Count; j++)
                    {
                        indegree[grafo[i][j]]--;
                    }
                    i = -1;
                }
            }

            if (cont == grafo.Length)
                return true;
            else
                return false;
        }

        static void Main(string[] args)
        {
            string[] linea = Console.ReadLine().Split();
            int vertices = int.Parse(linea[0]);
            int aristas = int.Parse(linea[1]);

            List<int>[] grafo = new List<int>[vertices];
            int[] indegree = new int[vertices];

            for (int i = 0; i < grafo.Length; i++)
            {
                grafo[i] = new List<int>();
            }

            for (int i = 0; i < aristas; i++)
            {
                linea = Console.ReadLine().Split();
                int a = int.Parse(linea[0]) - 1;
                int b = int.Parse(linea[1]) - 1;

                grafo[a].Add(b);
                indegree[b] += 1;
            }

            if (OrdenTopologico(grafo, indegree))
            {
                Console.WriteLine("True");
                for (int i = 0; i < ordenTopologico.Length; i++)
                {
                    Console.Write("{0} ", ordenTopologico[i] + 1);
                }
                Console.WriteLine();
            }
            else
                Console.WriteLine("False");
        }
    }
}
